cdap-notifications/src/main/java/co/cask/cdap/notifications/service/kafka/KafkaNotificationService.java

```diff
@@ -218,8 +219,9 @@ public void onReceived(Iterator<FetchedMessage> messages) {
           KafkaMessage decodedMessage = KafkaMessageCodec.decode(payload);
           try {
             LOG.trace("Decoded notification from Kafka: {}", decodedMessage);
-            notificationReceived(KafkaNotificationUtils.getMessageFeed(decodedMessage.getMessageKey()),
-                                 decodedMessage.getNotificationJson());
+            NotificationFeedId feedId =
+              NotificationFeedId.fromIdParts(Splitter.on('.').split(decodedMessage.getMessageKey()));
```

Comment:
if there are notifications in kafka that have not been consumed, they will be incompatible.